packages/block-library/src/embed/index.js

```diff
@@ -65,7 +69,7 @@ export function getEmbedEdit( title, icon ) {
 			const switchedPreview = this.props.preview && this.props.attributes.url !== prevProps.attributes.url;
 			const switchedURL = this.props.attributes.url !== prevProps.attributes.url;
 
-			if ( switchedURL && this.maybeSwitchBlock() ) {
+			if ( ( switchedURL || ( hasPreview && ! hadPreview ) ) && this.maybeSwitchBlock() ) {
```

Comment:
Try switching the block if the URL changes, or we get a preview. To switch to a WordPress block, we need the preview.